gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/jackson/ser/api/ApiSerializer.java

```diff
@@ -151,16 +151,18 @@ public void serialize(ApiEntity apiEntity, JsonGenerator jsonGenerator, Serializ
                 Set<MemberEntity> memberEntities = applicationContext.getBean(MembershipService.class).getMembersByReference(MembershipReferenceType.API, apiEntity.getId());
                 List<Member> members = new ArrayList<>(memberEntities == null ? 0 : memberEntities.size());
                 if (memberEntities != null) {
-                    memberEntities.forEach(m -> {
-                        UserEntity userEntity = applicationContext.getBean(UserService.class).findById(m.getId());
-                        if (userEntity != null) {
-                            Member member = new Member();
-                            member.setRoles(m.getRoles().stream().map(RoleEntity::getId).collect(Collectors.toList()));
-                            member.setSource(userEntity.getSource());
-                            member.setSourceId(userEntity.getSourceId());
-                            members.add(member);
-                        }
-                    });
+                    memberEntities.stream()
+                            .filter(m -> m.getType() == MembershipMemberType.USER)
+                            .forEach(m -> {
+                                UserEntity userEntity = applicationContext.getBean(UserService.class).findById(m.getId());
```

Comment:
I don't know if `getBean()` method is resource consuming but maybe we can get the UserService out of this loop ?